keps/sig-node/3857-rro-hostpath/README.md

```diff
@@ -0,0 +1,1046 @@
+      mountPropagation: None
+      # NEW
+      recursivelyReadOnly: BestEffort
+```
+
+`RecursivelyReadOnly` is (probably) specific to hostPath mounts, so it could be rather added to [`HostPathVolumeSource`](https://github.com/kubernetes/kubernetes/blob/v1.26.1/pkg/apis/core/types.go#L662-L666) struct.
+However, as the `VolumeMount` struct already has `MountPropagation`, `RecursivelyReadOnly` should be added to the `VolumeMount` struct too to keep the design consistent.
+
+Diff:
+```diff
+diff --git a/pkg/apis/core/types.go b/pkg/apis/core/types.go
+index 04ffd250b61..21ca8e3e006 100644
+--- a/pkg/apis/core/types.go
++++ b/pkg/apis/core/types.go
+@@ -1857,7 +1857,7 @@ type VolumeMount struct {
+        Name string
+        // Optional: Defaults to false (read-write).
+        // +optional
+-       ReadOnly bool
++       ReadOnly *bool
+        // Required. If the path is not an absolute path (e.g. some/path) it
+        // will be prepended with the appropriate root prefix for the operating
+        // system.  On Linux this is '/', on Windows this is 'C:\'.
+@@ -1878,6 +1878,9 @@ type VolumeMount struct {
+        // SubPathExpr and SubPath are mutually exclusive.
+        // +optional
+        SubPathExpr string
++       // When RecursivelyReadOnly is set to a non-nil value, ReadOnly is implied to be true.
++       // +optional
++       RecursivelyReadOnly *RecursivelyReadOnlyMode
+ }
+
+ // MountPropagationMode describes mount propagation.
+@@ -1905,6 +1908,18 @@ const (
+        MountPropagationBidirectional MountPropagationMode = "Bidirectional"
+ )
+
++// RecursivelyReadOnlyMode describes recursively-readonly mode.
++type RecursivelyReadOnlyMode string
++
++const (
++       // RecursivelyReadOnlyDisabled disables recursively-readonly mode.
++       RecursivelyReadOnlyDisabled RecursivelyReadOnlyMode = "Disabled"
++       // RecursivelyReadOnlyDisabled enables recursively-readonly mode if possible.
++       RecursivelyReadOnlyBestEffort RecursivelyReadOnlyMode = "BestEffort"
++       // RecursivelyReadOnlyDisabled enables recursively-readonly mode, or raise an error.
++       RecursivelyReadOnlyEnabled RecursivelyReadOnlyMode = "Enabled"
++)
++
+ // VolumeDevice describes a mapping of a raw block device within a container.
+ type VolumeDevice struct {
+        // name must match the name of a persistentVolumeClaim in the pod
+@@ -2505,6 +2520,9 @@ type ContainerStatus struct {
+        // +featureGate=InPlacePodVerticalScaling
+        // +optional
+        Resources *ResourceRequirements
++       // Status of volume mounts.
++       // +optional
++       VolumeMounts []VolumeMountStatus
+ }
+
+ // PodPhase is a label for the condition of a pod at the current time.
+@@ -2584,6 +2602,22 @@ const (
+        PodResizeStatusInfeasible PodResizeStatus = "Infeasible"
+ )
+
++// VolumeMountStatus shows status of volume mounts.
++type VolumeMountStatus struct {
++       // Corresponds to the name of the original VolumeMount.
++       Name string
```

Comment:
Mentioned elsewhere: Keep name, but the actual unique key is mountPath, so need to add that.

Reply:
Done